lib/LANraragi/Plugin/Hdoujin.pm

```diff
@@ -0,0 +1,114 @@
+package LANraragi::Plugin::Hdoujin;
+
+use strict;
+use warnings;
+
+#Plugins can freely use all Perl packages already installed on the system
+#Try however to restrain yourself to the ones already installed for LRR (see tools/cpanfile) to avoid extra installations by the end-user.
+use Mojo::JSON qw(from_json);
+
+#You can also use the LRR Internal API when fitting.
+use LANraragi::Model::Plugins;
+
+#Meta-information about your plugin.
+sub plugin_info {
+
+    return (
+        #Standard metadata
+        name      => "Hdoujin",
+        namespace => "Hdoujinplugin",
+        author    => "Pao",
+        version   => "0.1",
+        description =>
+"Collects metadata embedded into your archives by the Hdoujin .json files",
+    );
+
+}
+#Mandatory function to be implemented by your plugin
+sub get_tags {
+
+#LRR gives your plugin the recorded title/tags/thumbnail hash for the file, the filesystem path, and the custom arguments if available.
+    shift;
+    my ( $title, $tags, $thumbhash, $file, $oneshotarg, @args ) = @_;
+
+    my $logger = LANraragi::Utils::Generic::get_logger( "Hdoujin", "plugins" );
+
+    if ( LANraragi::Utils::Archive::is_file_in_archive( $file, "info.json" ) ) {
+
+        #Extract info.json
+        my $filepath = LANraragi::Utils::Archive::extract_file_from_archive( $file,
+            "info.json" );
+
+        #Open it
+        my $stringjson = "";
+
+        open( my $fh, '<:encoding(UTF-8)', $filepath )
+          or return ( error => "Could not open $filepath!" );
+
+        while ( my $row = <$fh> ) {
+            chomp $row;
+            $stringjson .= $row;
+        }
+
+        #Use Mojo::JSON to decode the string into a hash
+        my $hashjson = from_json $stringjson;
+
+        $logger->debug("Found and loaded the following JSON: $stringjson");
+
+        #Parse it
+        $tags = tags_from_Hdoujin_json($hashjson);
+
+        #Delete it
+        unlink $filepath;
+
+        #Return tags
+        $logger->info("Sending the following tags to LRR: $tags");
+        return ( tags => $tags );

```

Comment:
You can add the title here now that plugins support modifying the title of the archive. 
It'll fail gracefully on older versions of LRR (aka do nothing). 
```suggestion
        return ( tags => $tags,
                 title => $hashjson->{"manga_info"}->{"title"} );
```